task/task.go

```diff
@@ -48,6 +55,8 @@ func DoRetryWithTimeout(t func() (interface{}, bool, error), timeout, timeBefore
 		return out, err
 	case <-time.After(timeout):
 		quit <- true
-		return out, ErrTimedOut
+		return out, &ErrTimedOut{
```

Comment:
Isn't there a race condition here as there is no lock over `err` object?
You might get a timeout, while err is being set to nil.

Reply:
Good catch. Added a lock and also added UTs.